virttest/qemu_storage.py

```diff
@@ -93,13 +93,19 @@ def _get_image_meta(image, params, root_dir):
     if image_format == "qcow2" and image_encryption == "luks":
         meta["encrypt.key-secret"] = secret.aid
 
-    image_access = storage.ImageAccessInfo.access_info_define_by_params(image,
-                                                                        params)
-    if image_access is not None:
-        if image_access.auth is not None:
-            if image_access.storage_type == 'ceph':
+    auth_info = storage.StorageAuth.auth_info_define_by_params(image, params)
+    if auth_info is not None:
+        if auth_info.storage_type == 'ceph':
+            if auth_info.data:
                 # qemu-img needs secret object only for ceph access
-                meta['file']['password-secret'] = image_access.auth.aid
+                meta['file']['password-secret'] = auth_info.aid
+        elif auth_info.storage_type == 'iscsi-direct':
```

Comment:
It used to not handle the `iscsi-direct` part but looks needed now, would you please explain on that? thanks.

Reply:
That will be used in  source image's json repr for qemu-img command, e.g. create a snapshot:

/bin/qemu-img create --object secret,id=image1_encrypt0,data=redhat -f qcow2 -b 'json:{"file": {"target": "iqn.2019-09.com.example:zhencliu", "driver": "iscsi", "lun": 0, "portal": "10.66.10.26", "user": "redhat", "transport": "tcp", "password": "redhat", "initiator-name": "iqn.2008-11.org.linux-kvm"}, "driver": "luks", "key-secret": "image1_encrypt0"}' /home/kvm_autotest_root/images/snA.qcow2
@luckyh

Reply:
ok, I see, thanks for the information!